app/page.tsx

```diff
@@ -8,37 +8,126 @@ import {
-  const { data: markets, isLoading } = useQuery({
-    queryKey: ["getMarkets"],
+  const showClientUI = useShowClientUI();
+  const [selectedOption, setSelectedOption] = useState<Option>(
+    filterOptions[0]
+  );
+  const [search, setSearch] = useState("");
+
+  const { data, isLoading } = useQuery({
+    queryKey: ["getMarkets", search, selectedOption.orderBy],
     queryFn: async () =>
       getMarkets({
-        first: 10,
+        first: ITEMS_PER_PAGE,
         skip: 0,
-        orderBy: FixedProductMarketMaker_OrderBy.CreationTimestamp,
+        orderBy: selectedOption.orderBy,
         orderDirection: OrderDirection.Desc,
+        title_contains_nocase: search,
       }),
   });
 
+  const markets = data?.fixedProductMarketMakers;
+
   return (
-    <div className="px-6 mt-12 space-y-12 md:flex md:flex-col md:items-center">
-      <div>
-        <h1 className="mb-12 text-2xl font-semibold text-white">
-          🔮 All markets
-        </h1>
-        <div className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3  2xl:max-w-[1424px] 2xl:grid-cols-4 gap-4">
-          {isLoading
-            ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map(index => (
-                <LoadingCardMarket key={index} />
-              ))
-            : markets?.fixedProductMarketMakers.map(market => (
-                <Link key={market.id} href={`markets?id=${market.id}`}>
-                  <CardMarket market={market} />
-                </Link>
-              ))}
+    <div className="justify-center px-6 mt-12 space-y-12 md:px-10 lg:px-20 xl:px-40 md:flex md:flex-col md:items-center">
+      <div className="flex flex-col justify-between w-full gap-5 md:flex-row">
+        <h1 className="text-2xl font-semibold text-white">🔮 All markets</h1>
+
+        <div className="flex items-center space-x-2">
+          <Input
+            className="w-full md:w-72"
+            placeholder="Search market"
+            leftIcon="search"
+            onChange={(event) => setSearch(event.target.value)}
+            value={search}
+          />
+          {showClientUI ? (
+            <Popover>
+              <PopoverTrigger>
+                <Button variant="pastel" className="space-x-2">
+                  <p>{selectedOption.name}</p>
+                  <Icon name="chevron-down" />
+                </Button>
+              </PopoverTrigger>
+              <PopoverContent className="px-1 py-2">
+                {filterOptions.map((option) => {
+                  return (
+                    <div
+                      key={option.name}
+                      onClick={() => setSelectedOption(option)}
+                      className="flex items-center justify-start px-3 py-2 space-x-2 font-semibold cursor-pointer"
+                    >
+                      <Icon
+                        className={
+                          selectedOption.name !== option.name
+                            ? "invisible"
+                            : undefined
```

Comment:
Minor, non-blocking: 
This should behave the same
```suggestion
                        className={cx({
                          invisible: selectedOption.name !== option.name,
                        })}
```